### PR TITLE
[pruning]add dropout to list of supported activation functions

### PR DESCRIPTION
**Pull request by UsingtcNower**
Summary:
Add nn.Dropout and F.dropout to list of suppported activation functions

Test Plan:
```
python test/test_ao_sparsity.py -- TestBaseSparsifier
```
cc @kit1980  @andrewor14 @jcaip Could you please help to review this tiny change?

### CODE REVIEW

**Comment by UsingtcNower** [comment]
> suppported activation functions
Hi @jcaip thanks very much for your reply :) and thanks for providing the excellent torchao. I hava sync this tiny change to torchao with [here](https://github.com/pytorch/ao/pull/194), could you please help to review it?

**Comment by jcaip** [comment]
Closing as we're consolidating in torchao see [here](https://github.com/pytorch/ao/pull/194#issuecomment-2088926272) for context.
